config/config.go

```diff
@@ -244,12 +245,19 @@ func loadExtensions(v *viper.Viper, factories map[string]extension.Factory) (con
 		extensionCfg.SetType(typeStr)
 		extensionCfg.SetName(fullName)
 
-		// Now that the default config struct is created we can Unmarshal into it
-		// and it will apply user-defined config on top of the default.
-		if err := subViper.UnmarshalKey(key, extensionCfg); err != nil {
-			return nil, &configError{
-				code: errUnmarshalError,
-				msg:  fmt.Sprintf("error reading settings for extension type %q: %v", typeStr, err),
+		// Unmarsh only the subconfig for this extension.
+		sv := subViper.Sub(key)
+		if sv != nil {
```

Comment:
nit: `sv == nil` should never happen but if we must have the check then erroring on `sv == nil` is better (easier to read, harder to get code coverage in this case...)

Reply:
It happens, see comments https://github.com/open-telemetry/opentelemetry-service/pull/334#discussion_r323419676

Reply:
I see, so all is well then it goes with the default config, a comment to make that clear will be nice.